x-pack/plugins/data_visualizer/server/routes.ts

```diff
@@ -108,4 +106,50 @@ export function routes(coreSetup: CoreSetup<StartDeps, unknown>, logger: Logger)
         }
       }
     );
+
+  /**
+   * @apiGroup DataVisualizer
+   *
+   * @api {get} /internal/data_visualizer/inference/{inferenceId} Runs inference on a given inference endpoint with the provided input
+   * @apiName inference
+   * @apiDescription Runs inference on a given inference endpoint with the provided input.
+   */
+  router.versioned
+    .post({
+      path: '/internal/data_visualizer/inference/{inferenceId}',
+      access: 'internal',
+      options: {
+        tags: ['access:fileUpload:analyzeFile'],
```

Comment:
do we need an additional tag for inference?

Reply:
I don't think it's needed as es will be checking permissions.
The tags here are just for feature enablement.